protocol-units/bridge/contracts/src/AtomicBridgeInitiator.sol

```diff
@@ -46,12 +58,15 @@ contract AtomicBridgeInitiator is IAtomicBridgeInitiator, Initializable {
         }
         // If msg.value is greater than 0, convert ETH to WETH
         if (ethAmount > 0) weth.deposit{value: ethAmount}();
-        //Transfer WETH to this contract, revert if transfer fails
+        // Transfer WETH to this contract, revert if transfer fails
         if (wethAmount > 0) {
             if (!weth.transferFrom(originator, address(this), wethAmount)) revert WETHTransferFailed();
         }
 
-        nonce++; //increment the nonce
+        // Update the pool balance
+        poolBalance += totalAmount;
+
+        nonce++; // increment the nonce
         bridgeTransferId =
             keccak256(abi.encodePacked(originator, recipient, hashLock, timeLock, block.number, nonce));
```

Comment:
You should add a comment about the usage of the nonce here.

Reply:
That's been added.
```
        // The nonce is used to generate a unique bridge transfer id, without it 
        // we can't guarantee the uniqueness of the id.
```